pennylane/data/dataset.py

```diff
@@ -128,34 +128,46 @@ def attrs(self):
 
     @staticmethod
     def _read_file(filepath):
-        """Reading the data from a saved file. Returns a dictionary."""
+        """Read data from a saved file.
+
+        Returns:
+            A data value for non-standard datasets or full files, otherwise a dictionary
+        """
         with open(filepath, "rb") as f:
             compressed_pickle = f.read()
 
         depressed_pickle = zstd.decompress(compressed_pickle)
         return dill.loads(depressed_pickle)
 
-    def read(self, filepath):
+    def read(self, filepath, lazy=False):
         """Loads data from a saved file to the current dataset.
 
         Args:
-            filepath (string): the desired location and filename to load, e.g. './path/to/file/file_name.dat'.
+            filepath (string): The desired location and filename to load, e.g. './path/to/file/file_name.dat'.
+            lazy (bool): Indicates if only the key of the attribute should be saved to the Dataset instance
 
         **Example**
 
         >>> new_dataset = qml.data.Dataset(kw1 = 1, kw2 = '2', kw3 = [3])
         >>> new_dataset.read('./path/to/file/file_name.dat')
         """
-        data = self._read_file(filepath)
-        attribute = "full"  # set 'full' by default so non-standard datasets read keys and values
-        if self._is_standard:
-            attribute = self.__get_attribute_from_filename(filepath)
+        # set 'full' for non-standard datasets so they read keys too
+        attribute = self.__get_attribute_from_filename(filepath) if self._is_standard else "full"
         if attribute == "full":
+            data = self._read_file(filepath)
             for attr, value in data.items():
-                setattr(self, f"{attr}", value)
+                setattr(self, f"{attr}", None if lazy else value)
         else:
+            data = None if lazy else self._read_file(filepath)
             setattr(self, f"{attribute}", data)
 
+    @staticmethod
+    def _write_file(data, filepath, protocol=4):
```

Comment:
Short docstring might be useful. 
(specially since we have two _write_ functions)

Reply:
Agreed, but to be clear - methods with a leading `_` are protected, meaning users shouldn't really be using them anyway. Sadly, Python doesn't let us actually block users from using these methods directly, but they really shouldn't. I don't want to tempt them into using protected methods, but I'll add a little docstring.